ios/RCTMapboxGL/RCTMapboxGLManager.m

```diff
@@ -605,4 +605,85 @@ - (NSArray*)serializePacksArray:(NSArray<MGLOfflinePack*>*)packs
     }];
 }
 
+RCT_EXPORT_METHOD(queryRenderedFeatures:(nonnull NSNumber *)reactTag
+                  options:(NSDictionary *)options
+                  callback:(RCTResponseSenderBlock)callback)
+{
+    [_bridge.uiManager addUIBlock:^(RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTMapboxGL *> *viewRegistry) {
+        RCTMapboxGL *mapView = viewRegistry[reactTag];
+        if ([mapView isKindOfClass:[RCTMapboxGL class]]) {
+            NSDictionary *pointDict = options[@"point"];
+            NSDictionary *rect = options[@"rect"];
+            if (!pointDict) {
+                // TODO: figure out best way to error this. Should we be using RCTPromise rather than a callback?
+//                reject(@"invalid_arguments", @"queryRenderedFeatures(): screenCoordX and screenCoordY are required.", nil);
+                return;
+            }
+            NSNumber *screenCoordX = pointDict[@"screenCoordX"];
+            NSNumber *screenCoordY = pointDict[@"screenCoordY"];
+            NSSet<NSString *> *styleLayerIdentifiers = options[@"layers"];
```

Comment:
You're right. I should be pulling it out of `options` as an `NSArray`, then convert it to an `NSSet`. I just pushed some commits with that change.

> Probably better to convert it into an NSArray using `allObjects`.

You mean convert the NSArray into an NSSet as expected by `visibleFeaturesAtPoint`? Let me know if my most recent change to it looks good. I really appreciate your feedback!